.github/workflows/release.yaml

```diff
@@ -18,8 +18,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          # renovate: datasource=golang-version depName=go
-          go-version: 1.21.3
+          go-version-file: 'go.mod'
```

Comment:
I would suggest *not* to tie release workflows to `go.mod` version. Release workflows should always use the latest version, whereas I think that developer toolchains can lag behind within the same minor version and that's OK.

Reply:
In practice that means that the approach I'm exploring in the other linked PR is to have two renovate configs: One for go.mod following the `X.Y.0` releases and a different one for everything else following every `X.Y.Z` patch release.

Reply:
The actual implementation ended up different: There's a general Go version bump rule, then I added a specific rule for `go.mod` to disable patch updates for the toolchain version in the go.mod file.